week2/homework/src/index.js

```diff
@@ -1,3 +1,119 @@
-'use strict';
-
-// TODO: Write the homework code in this file
+'use strict';
+
+const fs = require('fs');
+const storageFile = 'todolist.json';
+
+function readFile() {
+  fs.readFile(storageFile, 'utf8', (err, list) => {
+    if (err) {
+      throw err;
+    }
+    console.log(`\nTodo list:\n${list}`);
+  });
+}
+
+function add() {
+  fs.readFile(storageFile, 'utf8', (err, list) => {
+    if (err) {
+      throw err;
+    }
+    list = JSON.parse(list);
+    list.push(todoObject);
+    fs.writeFile(storageFile, JSON.stringify(list, null, 4), err => {
+      if (err) throw err;
+    });
+    console.log(`'${todoObject}' has been added to the list.`);
+  });
+}
+
+function remove(index) {
+  fs.readFile(storageFile, 'utf8', (err, list) => {
+    if (err) {
+      throw err;
+    }
+    list = JSON.parse(list);
+    if (index > 0 && index <= list.length) {
+      list.splice(index - 1, 1);
+      fs.writeFile(storageFile, JSON.stringify(list, null, 4), err => {
+        if (err) throw err;
+        console.log(`Item at index ${index} has been terminated >=3`);

```

Comment:
2 levels deep in the nested callback hell 😢 I would really like that for week3 you use promises or async await